cpp/build-support/trim-boost.sh

```diff
@@ -32,12 +32,12 @@ set -eu
 : ${BOOST_URL:=https://dl.bintray.com/boostorg/release/${BOOST_VERSION}/source/${BOOST_FILE}.tar.gz}
 
 # Arrow tests require these
-BOOST_LIBS="system.hpp filesystem.hpp"
+BOOST_LIBS="process.hpp filesystem.hpp"
```

Comment:
I wasn't sure if it was safe to remove system.hpp because it is named explicitly in cmake and don't have time to track this down right now, so leaving a note to future self.

```suggestion
# TODO: is system still required? We declare `--with-libraries=filesystem,regex,system`
# in cmake, but there is no #include <boost/system.hpp> in our source anymore
BOOST_LIBS="system.hpp process.hpp filesystem.hpp"
```